src/main/java/edu/ie3/datamodel/io/source/csv/CsvIdCoordinateSource.java

```diff
@@ -159,6 +159,9 @@ protected Stream<Map<String, String>> buildStreamWithFieldsToAttributesMap() {
       // returning the original one
       Collection<Map<String, String>> allRows = csvRowFieldValueMapping(reader, headline);
 
+      // checking the column names for each row
+      allRows.forEach(row -> factory.checkForInvalidColumnNames(row.keySet()));
+
```

Comment:
I believe it'd be a lot more efficient to check the column headers only once (after line 154) and not repeatedly for every row read.